src/main/java/Duke.java

```diff
@@ -1,10 +1,251 @@
+        return item;
+    }
+
+    /**
+     * Formats the string that was converted from an array. Delete brackets and commas
+     *
+     * @param args the input string that was converted from an array
+     * @return the formatted string without brackets and commas
+     */
+    public static String getFormattedString(String args) {
+        String formattedString = args.replace(",", "")
+                .replace("]", "")
+                .replace("[", "")
+                .trim();
+        return formattedString;
+    }
+
+
+    /**
+     * Checks if the input string has an empty item after a command
+     *
+     * @param args the input string the user entered
+     * @return true if the item after the command is empty, false otherwise
+     */
+    public static boolean isEmptyItem(String args) {
+        if (getItem(args).equals("")) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * returns if there is a time entered for a deadline or event task
+     *
+     * @param args the input string the user entered
+     * @return true if there is a time entered, false otherwise
+     */
+    public static boolean canGetTime(String args) {
+        String description = getItem(args);
+        String[] time = description.substring(description.indexOf("/")).split(" ");
+        if (time.length < 2) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if the item from the user input is invalid for deadline and event tasks
+     *
+     * @param args the item to be checked
+     * @return true if the item does not contain a '/by' or does not have time after /by, false otherwise
+     */
+    public static boolean isInvalidDeadline(String args) {
+        if (!getItem(args).contains(BY_DIVIDER)) {
+            return true;
+        }
+        if (!canGetTime(args)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the item from the user input is invalid for deadline and event tasks
+     *
+     * @param args the item to be checked
+     * @return true if the item does not contain a '/at' or does not have time after /at, false otherwise
+     */
+    public static boolean isInvalidEvent(String args) {
+        if (!getItem(args).contains(AT_DIVIDER)) {
+            return true;
+        }
+        if (!canGetTime(args)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Reads the input and command entered by the user and execute task according to user command
+     *
+     * @param inputStr the full input the user entered
+     * @param command the command parsed from the input
+     * @param taskManager the task manager instance handling the tasks
+     * @param fileManager the file manager handling the files
+     *
+     **/
+    public static void executeCommand(String inputStr, String command, TaskManager taskManager, FileManager fileManager) throws DukeException {
```

Comment:
As a code quality measure and to improve the readability of your code, it is better to have methods with at most 30 LoC. Try to refactor and make this piece of code more modular

Reply:
Perhaps a design approach that you could consider is having a `Parser` class to parse the user inputs and get them in the desired format and a `Command` class that handles the execution of each command